### PR TITLE
Fix deprecated routing configuration

### PR DESCRIPTION
**Pull request by Soullivaneuh**
Cherry-pick of #1823 (cc @chriwup).

But Symfony 2.1 supports will be lost.

### CODE REVIEW

**Comment by chriwup** [comment]
Thanks for helping there @Soullivaneuh! tests are failing here too though.

**Comment by Soullivaneuh** [comment]
This is strange, tests are about `/{groupname}` without capital `N` but I especially get it on conflict and didn't changed it.

ping @stof

**Comment by XWB** [comment]
@Soullivaneuh If you're going to bump the Symfony requirement, perhaps 2.3 is even better.

**Comment by Soullivaneuh** [comment]
@XWB done.

**Comment by XWB** [comment]
Now lets find out why the tests are failing.

File: Tests/Routing/RoutingTest.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    public function loadRoutingProvider()
+    {
+        return array(
+            array('fos_user_change_password', '/change-password', array('GET', 'POST')),
+
+            array('fos_user_group_list', '/group/list', array('GET')),
+            array('fos_user_group_new', '/group/new', array('GET', 'POST')),
+            array('fos_user_group_show', '/group/{groupName}', array('GET')),
```
Comment (stloyd) [inline Tests/Routing/RoutingTest.php]: Variable name is not camel-cased in route definition.

**Comment by Soullivaneuh** [comment]
@stloyd @XWB fixed.

**Comment by Soullivaneuh** [comment]
Note: This PR should be merge on master too.

**Comment by stof** [comment]
@Soullivaneuh isn't the master branch already free of deprecations ?

**Comment by Soullivaneuh** [comment]
@stof was not. https://github.com/FriendsOfSymfony/FOSUserBundle/blob/d18641db1d28daf7ee27925ae56fd5cb04ba72af/Resources/config/routing/change_password.xml :-)

**Pull request merged**